File: Practica 3/CivitasJava/src/juegotexto/Controlador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package juegotexto;

/**
 *
 * @author juanc
 */
import civitasjava.CivitasJuego;
import civitasjava.Diario;
import civitasjava.GestionesInmobiliarias;
import civitasjava.Jugador;
import civitasjava.OperacionInmobiliaria;
import civitasjava.OperacionesJuego;
import civitasjava.Respuestas;
import civitasjava.SalidasCarcel;
import java.util.ArrayList;

public class Controlador {
    private CivitasJuego juego;
    private VistaTextual vista;
    
    public Controlador(CivitasJuego juego, VistaTextual vista) {
        this.juego = juego;
        this.vista = vista;
    }
    
    public void juega() {
       
        vista.setCivitasJuego(juego);
        
        while(!juego.finalDelJuego()) {
            
            vista.pausa();
            OperacionesJuego operacion = vista.juegoModel.siguientePaso(); 
            this.vista.mostrarSiguienteOperacion(operacion); 
            
            if (operacion != OperacionesJuego.PASAR_TURNO){ 
                this.vista.mostrarEventos();
                this.vista.actualizarVista();
            }
    
            
            if (!juego.finalDelJuego()) {
                switch(operacion) {
                    case COMPRAR:
                        Respuestas respuesta = vista.comprar();
                        
                        if (respuesta == Respuestas.SI) 
                            this.juego.comprar();
                            
                        this.juego.siguientePasoCompletado(operacion);
                        
                        break;
                        
                    case GESTIONAR:
                        vista.gestionar();
                        GestionesInmobiliarias gestion = GestionesInmobiliarias.values()[vista.getGestion()];
                        OperacionInmobiliaria op_inmobiliaria = new OperacionInmobiliaria(gestion,vista.getPropiedad());
                        
                        switch (op_inmobiliaria.getGestion()) {
                            case CANCELAR_HIPOTECA:
                                this.juego.cancelarHipoteca(vista.getPropiedad());
                                break;
                            case CONSTRUIR_CASA:
                                this.juego.construirCasa(vista.getPropiedad());
                                break;
                            case CONSTRUIR_HOTEL:
                                this.juego.construirHotel(vista.getPropiedad());
                                break;
                            case HIPOTECAR:
                                this.juego.hipotecar(vista.getPropiedad());
                                break;
                            case TERMINAR:
                                juego.siguientePasoCompletado(operacion);
                                break;
                            case VENDER:
                                this.juego.vender(vista.getPropiedad());
                                break;
                        }
                        break;
                    case SALIR_CARCEL:
                        SalidasCarcel salida = vista.salirCarcel();
                        
                        if (salida == SalidasCarcel.PAGANDO) {
                            this.juego.salirCarcelPagando();
                        }
                        else if (salida == SalidasCarcel.TIRANDO) {
                            this.juego.salirCarcelTirando();
                        }
                        
                        this.juego.siguientePasoCompletado(operacion);
                        break;
                }
            }
            
        }
        
        System.out.println("\n---RANKING---ºn");
        ArrayList<Jugador> ranking = this.juego.ranking();

        for (int i = 0; i < ranking.size(); i++) {
            System.out.println(ranking.get(i) + "\n");
        }
        
    }
}

File: Practica 4/CivitasRuby4/CivitasRuby/pkg/RubyApplication1-0.0.1/lib/controlador.rb
#encoding:utf-8

module Civitas
  class Controlador
    def initialize(juego,vista)
      @juego = juego
      @vista = vista
    end
    
    def juega
    @vista.set_civitas_juego(@juego)
    
    while (!@juego.final_del_juego) do
      @vista.pausa
      operacion = @vista.juegoModel.siguiente_paso
      
      if (operacion != Operaciones_juego::PASAR_TURNO)
        @vista.mostrar_eventos
        #@vista.actualizar_vista
      else
        @vista.mostrar_estado("Pasando turno")
      end
      
      if (!@juego.final_del_juego)
        case operacion
          when Operaciones_juego::COMPRAR
            respuesta = @vista.comprar
          
            if (respuesta == Respuestas::SI)
              @juego.comprar
            end
            @juego.siguiente_paso_completado(operacion)
            
          when Operaciones_juego::GESTIONAR
            @vista.gestionar
            lista_gestiones = [Gestiones_inmobiliarias::VENDER,Gestiones_inmobiliarias::HIPOTECAR,
                Gestiones_inmobiliarias::CANCELAR_HIPOTECA,Gestiones_inmobiliarias::CONSTRUIR_CASA,
                Gestiones_inmobiliarias::CONSTRUIR_HOTEL,Gestiones_inmobiliarias::TERMINAR]
            gestion = lista_gestiones[@vista.iGestion]
            
          case gestion
            when Gestiones_inmobiliarias::CANCELAR_HIPOTECA
                @juego.cancelar_hipoteca(@vista.iPropiedad)
              when Gestiones_inmobiliarias::CONSTRUIR_CASA
                @juego.construir_casa(@vista.iPropiedad)
              when Gestiones_inmobiliarias::CONSTRUIR_HOTEL
                @juego.construir_hotel(@vista.iPropiedad)
              when Gestiones_inmobiliarias::HIPOTECAR
                @juego.hipotecar(@vista.iPropiedad)
              when Gestiones_inmobiliarias::TERMINAR
                @juego.siguiente_paso_completado(operacion)
              when Gestiones_inmobiliarias::VENDER
                @juego.vender(@vista.iPropiedad)
            end
          when Operaciones_juego::SALIR_CARCEL
            salida = @vista.salir_carcel

            if (salida == Salidas_carcel::PAGANDO)
              @juego.salir_carcel_pagando
            elsif (salida == Salidas_carcel::TIRANDO)
              @juego.salir_carcel_tirando
            end

            @juego.siguiente_paso_completado(operacion)
         end
       end 
     end
     
    puts "\n---RANKING---"
      ranking = @juego.ranking

      for i in 0..@juego.ranking-1
        puts ranking.at(i) + "\n"
      end
   end
  end
 end
